server.cc

```diff
@@ -240,8 +249,8 @@ class server_handler: public serverIf {
         for (auto method: valid_methods) {
             methods.push_back(method);
         }
-        
-        set_fault(methods, random, err_no, probability,
+
+        set_fault(methods, random, err_nos, probability,
```

Comment:
ditto, `&` coding style, and consider moving a vector rather than passing a reference around and eventually copy it, just to throw away the original vector.